Cohesity First Draft

jkitchens32 opened the pull request:
## What Changed?

First draft of Cohesity Guide

jkitchens32 commented on advocacy_docs/partner_docs/CohesityBackupandRecover/02-PartnerInformation.mdx:
```diff
@@ -0,0 +1,12 @@
+| &nbsp;     | &nbsp; |
+| ----------- | ----------- |
+| **Partner Name**      | Cohesity       |
+| **Partner Product**   |  Cohesity Postgres Connector Agent      |
```
The specific product that we tested was called the Cohesity Postgres Connector Agent, which is what I got from the Cohesity tech side.

jkitchens32 commented:
@sethijs I have made picture edits to clean up the screenshots, edits to the wording as you discussed.

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/03-SolutionSummary.mdx:
```diff
@@ -0,0 +1,19 @@
+- EDB Postgres Advanced Server
+- Cohesity Backup Appliance
+- Cohesity PostgreSQL connector
+- Cohesity Connector Agent
```
"Connector" in "Cohesity PostgreSQL connector" should be capitalized

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/03-SolutionSummary.mdx:
```diff
@@ -0,0 +1,19 @@
+- Cohesity Connector Agent
+
+
+The diagram below shows how you can integrate the Cohesity Platform with your EDB Postgres Advanced Server deployments using the Cohesity PostgreSQL connector and Connector Agent.
```
"Platform" should be lowercase and "Connector" in "Cohesity PostgreSQL connector" should be uppercase.

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/04-ConfiguringCohesityBackupandRecoveryforPostgres.mdx:
```diff
@@ -0,0 +1,76 @@
+---
+title: 'Configuring Cohesity for Postgres'
+description: 'Walkthrough on configuring Cohesity Postgres Connector Agent'
```
Should be "Walkthrough of configuring Cohesity" or whatever product name Lina gives us.

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/04-ConfiguringCohesityConnectorAgentforEDBPostgreSQL.mdx:
```diff
@@ -0,0 +1,76 @@
+
+   ![Cohesity Dashboard](Images/CohesityDashboard.png)
+
+4. Register your EDB Postgres Advanced Server database. From the main Dashboard screen select Data Protection and then select Sources on the left-hand side.
```
Use code syntax on "Data Protection" and "Sources"

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/04-ConfiguringCohesityConnectorAgentforEDBPostgreSQL.mdx:
```diff
@@ -0,0 +1,76 @@
+
+   ![Cohesity Register Universal Data Adapter](Images/CohesityRegisterUniversalDataAdapter.png)
+
+7. Once on the Register Universal Data Adapter screen input the specific source information in order for the source to be protected.
```
Put a comma between "screen" and "input"

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/04-ConfiguringCohesityConnectorAgentforEDBPostgreSQL.mdx:
```diff
@@ -0,0 +1,76 @@
+   ![Cohesity Register Universal Data Adapter](Images/CohesityRegisterUniversalDataAdapter.png)
+
+7. Once on the Register Universal Data Adapter screen input the specific source information in order for the source to be protected.
+- **Source Type**: For EDB Postgres Advanced Server, Source Type will be PostgreSQL
```
End all the definitions for the bullets with a period.

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/04-ConfiguringCohesityConnectorAgentforEDBPostgreSQL.mdx:
```diff
@@ -0,0 +1,76 @@
+7. Once on the Register Universal Data Adapter screen input the specific source information in order for the source to be protected.
+- **Source Type**: For EDB Postgres Advanced Server, Source Type will be PostgreSQL
+- **Hostnames/IP Addresses**: This will be either the Hostname or IP Address of the specific source to register
+- **Script Directory Path**: This path is defined during the installation of the Cohesity Postgres Agent.  It will be displayed at the bottom of the installation scripts when the download is complete and it is the location of where the scripts are located: /opt/cohesity/postgres/scripts
```
Use code syntax for "/opt/cohesity/postgres/scripts"

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/04-ConfiguringCohesityConnectorAgentforEDBPostgreSQL.mdx:
```diff
@@ -0,0 +1,76 @@
+- **Hostnames/IP Addresses**: This will be either the Hostname or IP Address of the specific source to register
+- **Script Directory Path**: This path is defined during the installation of the Cohesity Postgres Agent.  It will be displayed at the bottom of the installation scripts when the download is complete and it is the location of where the scripts are located: /opt/cohesity/postgres/scripts
+- In the **App Authentication** section enter the admin username and password for the user who has admin privileges on your database to perform a backup.
+      !!! Note- Instead of password-based authentication, if you want to use kerberos authentication, then leave the username and password fields blank.
```
Note formatting is incorrect

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/04-ConfiguringCohesityConnectorAgentforEDBPostgreSQL.mdx:
```diff
@@ -0,0 +1,76 @@
+- **Source Type**: For EDB Postgres Advanced Server, Source Type will be PostgreSQL
+- **Hostnames/IP Addresses**: This will be either the Hostname or IP Address of the specific source to register
+- **Script Directory Path**: This path is defined during the installation of the Cohesity Postgres Agent.  It will be displayed at the bottom of the installation scripts when the download is complete and it is the location of where the scripts are located: /opt/cohesity/postgres/scripts
+- In the **App Authentication** section enter the admin username and password for the user who has admin privileges on your database to perform a backup.
```
Use a comma between "section" and "enter"

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/04-ConfiguringCohesityConnectorAgentforEDBPostgreSQL.mdx:
```diff
@@ -0,0 +1,76 @@
+   ![Cohesity Global Source Registration](Images/CohesityGlobalSourceRegistration.png)
+
+9. In the Global Source Registration Arguments field enter the following with the same syntax:
+--source-name=cohesitypostgres --postgres-user=enterprisedb --host=172.24.113.13 --port=5444 --pg-bin=/usr/edb/as14/bin
```
Use code syntax for "--source-name=cohesitypostgres --postgres-user=enterprisedb --host=172.24.113.13 --port=5444 --pg-bin=/usr/edb/as14/bin"

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/04-ConfiguringCohesityConnectorAgentforEDBPostgreSQL.mdx:
```diff
@@ -0,0 +1,76 @@
+- -- **postgres-user:** Specifies the user account to start the database
+- -- **host:** The IP address of the EDB Postgres Advanced Server instance
+- -- **port:** The option to specify which port the EDB Postgres Advanced Server is listening on.  If it is not specified the default port 5432 is used
+- -- **pg-bin:** The directory path to the EDB Postgres Advanced Server binaries, for example psql.
```
Get rid of the period on line 74, we want to keep the formatting consistent.

jkitchens32 replied:
Added periods to all.

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/05-UsingCohesityConnectorAgentforEDBPostgres.mdx:
```diff
@@ -0,0 +1,175 @@
+---
+title: 'Using Cohesity Connector Agent for EDB Postgres with EDB Postgres Advanced Server'
+description: 'Walkthrough of taking a backup and restore with the Cohesity Connector Agent for EDB Postgres scenario'
```
Description needs to be simple and consistent with other guides. Change to: 

"Walkthroughs of multiple Cohesity Connector Agent for EDB Postgres usage scenarios"

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/05-UsingCohesityConnectorAgentforEDBPostgres.mdx:
```diff
@@ -0,0 +1,175 @@
+
+You can restore an EDB Postgres Advanced Server instance to the same EDB Postgres Advanced server or to a different server.  At the moment the restores are full database restores, granular restores are not supported yet. 
+
+Recovery Script Arguments are an important piece of the recovery process, they define script parameters for your EDB Postgres Advanced Server recovery.  The restore script argument is: - -target-dir= with the second half being the path on the target cluster where you want to restore the data.
```
Use code syntax on "- -target-dir="

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/05-UsingCohesityConnectorAgentforEDBPostgres.mdx:
```diff
@@ -0,0 +1,175 @@
+### EDB Postgres Advanced Server Database Prerequisites Before Recovery Initiation
+
+Before you can recover your EDB Postgres Advanced Server instance, you will need to make sure that the server is not running.  Verify that your server is stopped, if it is running, stop it.
+    **For Example:** Log in to your machine as a Root user, and with systemctl commands first check the status of your instance to determine whether it is up, systemctl status edb-as-(version), and if it is running use systemctl stop edb-as-(version) to stop your server for the recovery to take place.
```
Do a line of space before the "For Example", it will come out looking cleaner.

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/05-UsingCohesityConnectorAgentforEDBPostgres.mdx:
```diff
@@ -0,0 +1,175 @@
+8. In the Recovery Options area, you will configure the following options:
+- **Mounts/VIPs**: The number of IP's to be used
+- **Concurrency**: The number of parallel streams to read data from Cohesity storage
+- **Recovery Script Arguments**: The path on the target cluster for where you want to restore data.  An example looks like: - - target-dir=/var/lib/edb/as13/data
```
Use code syntax on "- - target-dir=/var/lib/edb/as13/data"

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/05-UsingCohesityConnectorAgentforEDBPostgres.mdx:
```diff
@@ -0,0 +1,175 @@
+   ![Cohesity Recovery Logs](Images/CohesityRecoveryLogs.png)
+
+11. After the Restore has completed successfully, login to your EDB Postgres Advanced Server instance and check that the restore operation successfully recovered the data.
+    **Example:** Database “as13backup” dropped before the recovery:
```
Do a line of space before the example, it will come out cleaner

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/05-UsingCohesityConnectorAgentforEDBPostgres.mdx:
```diff
@@ -0,0 +1,175 @@
+
+   ![Restored Database](Images/CohesityEDBRestoredDatabase.png)
+
+
```
Add another line of space

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/05-UsingCohesityConnectorAgentforEDBPostgres.mdx:
````diff
@@ -0,0 +1,175 @@
+(4 rows)_
+
+```
+
````
Add another line of space

sethijs commented on advocacy_docs/partner_docs/CohesityBackupandRecover/06-CertificationEnvironment.mdx:
```diff
@@ -0,0 +1,10 @@
+---
+title: 'Certification Environment'
+description: 'Overview of the certification environment used in the certification of Cohesity'
```
Add the product name they asked us to use

jkitchens32 commented:
@sethijs I have gone through and made the name changes per the email conversation earlier.

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/02-PartnerInformation.mdx:
```diff
@@ -0,0 +1,12 @@
+| **Partner Product**   |  Cohesity DataProtect for PostgreSQL      |
+| **Web Site**      | https://www.cohesity.com/     |
+| **Version**   | Cohesity 6.6.0d       |
+| **Product Description**      | Cohesity simplifies database backup and recovery by eliminating multiple point products with a single software for on-premises or as-a-Service deployment. Protect and manage EDB Postgres Advanced Server and PostgreSQL data sources from a single global UI. |
```
Need a new Product Description.  That one is not working.  Mostly the second sentence.  I am sure we are not protecting and managing the database.

jkitchens32 replied:
"Simplify protection of EDB Postgres Advanced Server and PostgreSQL data sources from a single global UI." Do we want to get into a lot of detail or keep this more simple?

jkitchens32 replied:
"Cohesity simplifies database backup and recovery by eliminating multiple point products with a single software for on-premises or as-a-Service deployment. Streamline protection of EDB Postgres Advanced Server and PostgreSQL data sources from a single global UI."

Current, updated product description.  @scottatchue

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/03-SolutionSummary.mdx:
```diff
@@ -0,0 +1,19 @@
+title: 'Solution Summary'
+description: 'Brief explanation of the solution and its purpose'
+---
+Using Cohesity DataProtect for PostgreSQL, you can backup and restore your EDB Postgres Advanced Server and PostgreSQL databases, set specific backup policies for them, and keep track of the backup and restore runs. Cohesity DataProtect for PostgreSQL provides peace of mind for database owners with its nontraditional approaches to backups.  The Cohesity platform natively integrates with PostgreSQ ...
```
Change Using to "With"
With Cohesity DataProtect ....

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/03-SolutionSummary.mdx:
```diff
@@ -0,0 +1,19 @@
+- Cohesity Connector Agent
+
+
+The diagram below shows how you can integrate the Cohesity platform with your EDB Postgres Advanced Server and PostgreSQL deployments using Cohesity DataProtect for PostgreSQL.
```
Diagram is not showing how you can integrate.
I think we can remove the entire sentence.  See Veritas solution summary as an example.

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/04-ConfiguringCohesityDataProtectforPostgreSQL.mdx:
```diff
@@ -0,0 +1,81 @@
+
+- A running EDB Postgres Advanced Server instance
+- Cohesity Backup Appliance installed and running on VMware
+- Cohesity agents installed on the same machine as your EDB Postgres Advanced Server database
```
should just name the agents and not say agents.   So Linux and PostgreSQL agentes.

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/03-SolutionSummary.mdx:
```diff
@@ -0,0 +1,19 @@
+
+- EDB Postgres Advanced Server or PostgreSQL database
+- Cohesity Backup Appliance
+- Cohesity PostgreSQL Connector
```
These names (which ever is correct)

Cohesity Connector Agent
Cohesity PostgreSQL Connector
should match what is in Prerequisites section.
- Cohesity Linux Agent 
- Cohesity PostgreSQL Agent

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/04-ConfiguringCohesityDataProtectforPostgreSQL.mdx:
```diff
@@ -0,0 +1,81 @@
+
+   ![Cohesity Dashboard](Images/CohesityDashboard.png)
+
+4. Register your EDB Postgres Advanced Server database. From the main Dashboard screen select `Data Protection` and then select `Sources` on the left-hand side.
```
This section is Registering your EPAS.
Can delete "Register your EDB Postgres Advanced Server database."

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/04-ConfiguringCohesityDataProtectforPostgreSQL.mdx:
```diff
@@ -0,0 +1,81 @@
+7. Once on the Register Universal Data Adapter screen, input the specific source information in order for the source to be protected.
+
+- **Source Type**: For EDB Postgres Advanced Server, Source Type will be PostgreSQL.
+- **Hostnames/IP Addresses**: This will be either the Hostname or IP Address of the specific source to register.
```
instead of source to register can we say database?

jkitchens32 replied:
Yes that will work since we defined the source above

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/04-ConfiguringCohesityDataProtectforPostgreSQL.mdx:
```diff
@@ -0,0 +1,81 @@
+
+- **Source Type**: For EDB Postgres Advanced Server, Source Type will be PostgreSQL.
+- **Hostnames/IP Addresses**: This will be either the Hostname or IP Address of the specific source to register.
+- **Script Directory Path**: This path is defined during the installation of the Cohesity Postgres Agent.  It will be displayed at the bottom of the installation scripts when the download is complete, and it is the location of where the scripts are located: `/opt/cohesity/postgres/scripts`.
```
In the prereq section should we add this path?   Postgres Agent scripts path.

jkitchens32 replied:
I think we can add it as a planning heads up for the customer who downloads Cohesity.

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/04-ConfiguringCohesityDataProtectforPostgreSQL.mdx:
```diff
@@ -0,0 +1,81 @@
+
+   ![Cohesity Universal Data Adapter Information](Images/CohesityUniversalDataAdapterInformation.png)
+
+8. Select `Custom Options`
```
Should the be "Selct" or "Turn On",  or "Enable"?

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/04-ConfiguringCohesityDataProtectforPostgreSQL.mdx:
```diff
@@ -0,0 +1,81 @@
+
+!!! Note Instead of password-based authentication, if you want to use kerberos authentication, then leave the username and password fields blank.
+!!!
+- Select `Custom Options` and proceed with the information below for your EDB Postgres Advanced Server source.
```
can delete this line.  Covered in step 8.

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/04-ConfiguringCohesityDataProtectforPostgreSQL.mdx:
```diff
@@ -0,0 +1,81 @@
+- -- **source-name:** A unique name to identify the EDB Postgres Advanced Server cluster.
+- -- **postgres-user:** Specifies the user account to start the database.
+- -- **host:** The IP address of the EDB Postgres Advanced Server instance.
+- -- **port:** The option to specify which port the EDB Postgres Advanced Server is listening on.  If it is not specified the default port 5432 is used.
```
don't need "The option to specify which port"
change
The port the EDB ....
This way it is written just like host.

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/04-ConfiguringCohesityDataProtectforPostgreSQL.mdx:
```diff
@@ -0,0 +1,81 @@
+- -- **port:** The option to specify which port the EDB Postgres Advanced Server is listening on.  If it is not specified the default port 5432 is used.
+- -- **pg-bin:** The directory path to the EDB Postgres Advanced Server binaries, for example psql.
+
+10. Click `Register` and your source will be created and ready to backup.
```
Should we add "Configuring EDB Postgres Advanced Server" section like we have in Veritas guide?  This needs to be done for backup to work.

jkitchens32 replied:
I did see this note digging deeper in the Cohesity docs- "If you have not set archive_mode=on, the PostgreSQL database is restarted to set archive_mode=on."

So I can add the section you mentioned with this as a note if you think that is best?

scottatchue replied:
That sounds good.  Lets do that.

jkitchens32 replied:
Got it I'll get that added.

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/05-UsingCohesityDataProtectforPostgreSQL.mdx:
```diff
@@ -0,0 +1,182 @@
+### Create a Standard Policy
+1. In the Cohesity Dashboard, select `Data Protection`, then select `Policies`. When you are brought to the Policy page click on `Create Policy` located at the top right of the page.
+
+2. Specify a Policy Name for your protection policy 
```
period missing at end

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/05-UsingCohesityDataProtectforPostgreSQL.mdx:
```diff
@@ -0,0 +1,182 @@
+2. Specify a Policy Name for your protection policy 
+
+   ![Cohesity Create Protection Policy](Images/CohesityCreateProtectionPolicy.png)
+
```
Need a step added

Click Create

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/05-UsingCohesityDataProtectforPostgreSQL.mdx:
```diff
@@ -0,0 +1,182 @@
+
+   ![Cohesity Create Protection Policy](Images/CohesityCreateProtectionPolicy.png)
+
+3. After you have named your protection policy, you then specify all the options for when backups are captured. 
```
Change step 3 to
Specify all the options for when you want backups captured.

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/05-UsingCohesityDataProtectforPostgreSQL.mdx:
```diff
@@ -0,0 +1,182 @@
+description: 'Walkthroughs of multiple Cohesity DataProtect for PostgreSQL usage scenarios'
+---
+This section features use cases that show how backups are taken and restored from EDB Postgres Advanced Server using Cohesity DataProtect for PostgreSQL.
+
```
The following should all be a heading that is clickable under the "ON THIS PAGE" section just like "Cohesity Recovery of EDB Postgres Advanced Server Data" is
Create a Standard Policy
Protect an EDB Postgres Advanced Server Source with a Policy
Restore an EDB Postgres Advanced Server Instance from a Cohesity Protection Run
EDB Postgres Advanced Server Database Prerequisites Before Recovery Initiation
Cohesity Recovery of EDB Postgres Advanced Server Data

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/05-UsingCohesityDataProtectforPostgreSQL.mdx:
```diff
@@ -0,0 +1,182 @@
+
+### Protect an EDB Postgres Advanced Server Source with a Policy
+
+1. From the Cohesity Dashboard select `Data Protection` then select `Protection`. Select `Protect` on the top-right of the page and select `Universal Data Adapter`.
```
should this be multiple steps?  I see a lot of instructions

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/05-UsingCohesityDataProtectforPostgreSQL.mdx:
```diff
@@ -0,0 +1,182 @@
+7. Select `Replace Original` or `Recover to a New Object` depending on your requirement.
+
+8. In the Recovery Options area, you will configure the following options:
+
```
why are the bullets spaced?  we don't do this in other areas.

jkitchens32 replied:
We were having a line break issue that was a technical thing with the Markdown code. DeeDee and I went over what was causing it and I let Jasmer know in case we need to add this to any documentation for formatting.

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/05-UsingCohesityDataProtectforPostgreSQL.mdx:
```diff
@@ -0,0 +1,182 @@
+
+9. Click `Start Recovery` and your recovery process will be under way. 
+
+10. You can see a Status message on the main Recoveries page once the process has completed.  Click on the `Recover` task, then click `Show Subtasks` to view the logs from the recovery run.
```
2 steps?

jkitchens32 commented:
I have addressed the comments I have seen, fixing some formatting and getting with DeeDee to figure out some Markdown Issues (IE the bullet spacing). I have requested some final reviews from @sethijs and @scottatchue  to see if these changes are satisfactory.

scottatchue commented on advocacy_docs/partner_docs/CohesityDataProtectforPostgreSQL/04-ConfiguringCohesityDataProtectforPostgreSQL.mdx:
```diff
@@ -0,0 +1,102 @@
+- Login information for the Cohesity Web interface
+- Credentials that are used to access the EDB Postgres Advanced Server database
+- Java 1.8 or higher version is installed on the PostgreSQL nodes
+- PostgreSQL Agent scripts path, `/opt/cohesity/postgres/scripts`
```
is this Cohesity agent?

jkitchens32 replied:
Yes so sorry somehow left out the word Cohesity.
